File: ios-native/WordWheel/Models/Difficulty.swift
import SwiftUI

/// Difficulty buckets shown to the player at level milestones (10, 20,
/// 40, 60, 80). Mirrors the Android `Difficulty` enum.
enum Difficulty {
    case easy
    case medium
    case hard
    case expert
    case master
    case legend

    var displayName: String {
        switch self {
        case .easy:   return "Easy"
        case .medium: return "Medium"
        case .hard:   return "Hard"
        case .expert: return "Expert"
        case .master: return "Master"
        case .legend: return "Legend"
        }
    }

    var tagline: String {
        switch self {
        case .easy:   return "A gentle warm-up."
        case .medium: return "Things start mixing up."
        case .hard:   return "Crosswords get bigger."
        case .expert: return "Multi-word grids and rare letters."
        case .master: return "8-letter spines from here on."
        case .legend: return "Push through to 100."
        }
    }

    var accent: Color {
        switch self {
        case .easy:   return Color(hex: 0xFF32C850)
        case .medium: return Color(hex: 0xFF50A0E6)
        case .hard:   return Color(hex: 0xFFFFB400)
        case .expert: return Color(hex: 0xFFE65050)
        case .master: return Color(hex: 0xFFB450E6)
        case .legend: return Color(hex: 0xFFFFDC50)
        }
    }

    /// The tier the player just unlocked by completing `level`. Returns
    /// nil when no milestone fires.
    static func milestone(for completedLevel: Int) -> Difficulty? {
        switch completedLevel {
        case 10: return .medium
        case 20: return .hard
        case 40: return .expert
        case 60: return .master
        case 80: return .legend
        default: return nil
        }
    }
}

File: android-native/app/src/main/kotlin/com/wheelword/game/Difficulty.kt
package com.wheelword.game

/**
 * Difficulty buckets shown to the player at level milestones (10, 20,
 * 30 …). The text + accent colour bubble pops via [DifficultyDialog]
 * after the corresponding level finishes, signalling the next tier.
 *
 * This is a pure presentation thing — gameplay difficulty is already
 * dictated by the procedurally generated puzzle complexity per level.
 */
enum class Difficulty(
    val displayName: String,
    val tagline: String,
    val accentHex: Long,
) {
    EASY(    "Easy",       "A gentle warm-up.",                   0xFF32C850),
    MEDIUM(  "Medium",     "Things start mixing up.",             0xFF50A0E6),
    HARD(    "Hard",       "Crosswords get bigger.",              0xFFFFB400),
    EXPERT(  "Expert",     "Multi-word grids and rare letters.",  0xFFE65050),
    MASTER(  "Master",     "8-letter spines from here on.",       0xFFB450E6),
    LEGEND(  "Legend",     "Push through to 100.",                0xFFFFDC50),
    ;
    companion object {
        /**
         * Tier the player just unlocked by completing [completedLevel].
         * Returns null when no milestone fires (i.e. levels 1–9, 11–19, …).
         */
        fun milestoneFor(completedLevel: Int): Difficulty? = when (completedLevel) {
            10 -> MEDIUM
            20 -> HARD
            40 -> EXPERT
            60 -> MASTER
            80 -> LEGEND
            else -> null
        }
    }
}
